index.js

```diff
@@ -1,88 +1,119 @@
 // ***** Question 1 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
-
-// console.log(counter) 
+console.log("%cQuestion 1", "color: red")
+let counter = 1
+counter = 2
+console.log(counter)
 // // => 2
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red")
 
 
 // ***** Question 2 *****
 // *** Uncomment the lines below to test
-// *** after testing, comment the line causing the error back in 
+// *** after testing, comment the line causing the error back in
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
+console.log("%cQuestion 2", "color: red")
+const name = "James"
 
-// console.log(name)
+console.log(name)
 // => "Raffy"
-// name = "Not Raffy" 
+// name = "Not Raffy"
 // => TypeError
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red")
 
 
 
-// ***** Question 3 *****
-// *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+console.log("%cQuestion 3", "color: red")
+
+function drinkWater(currentThirstLevel) {
+  console.log ("Man I sure am thirsty")
+  currentThirstLevel = currentThirstLevel - 1
+  console.log("Ahh that hits the spot")
+  return (currentThirstLevel)
+}
+
 
-// console.log(drinkWater(12))
+console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red")
 
 
 
-// ***** Question 4 *****
-// *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red")
 
-// console.log(sameSameButDifferent(5, 5)) 
+function sameSameButDifferent(num, maybeNum) {
+  if (num == parseInt(maybeNum)) {
+    if (typeof(num) == typeof(maybeNum)) {
+      return ("same same")
+    } else {
+      return ("same same (but different)")
+    }
+  }
+  if (num != parseInt(maybeNum)) {
+      return ("different")
+  }
+}
+console.log(sameSameButDifferent(5, 5))
```

Comment:
This works! Check out this [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness) about equality comparison.

- Your first if-statement is checking if `num` is loosely equivalent to `maybeNum` ([`==`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality)). You could get away with not using `parseInt` to convert `maybeNum` in that conditional.
- Check out another possible solution using the [strict equality operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators#identity). This solution assumes `num` is always a Number data type.
```js
function sameSameButDifferent(num, maybeNum) {
  if (num === maybeNum) {
    return 'same same'
  }
  else if (num === parseInt(maybeNum)) {
    return 'same same (but different)'
  }
  else {
    return 'different'
  }
}
```

- Check out this solution where we don't assume `num` is a Number data type. The only change is in the first conditional statement
```js
function sameSameButDifferent(num, maybeNum) {
  if (typeof num === 'number' && num === maybeNum) {
    return 'same same'
  }
  else if (num === parseInt(maybeNum)) {
    return 'same same (but different)'
  }
  else {
    return 'different'
  }
}
```